tests/dependency_analyzer/expected_results/skipdep_micronaut-projects_micronaut-test.json

```diff
@@ -3,7 +3,7 @@
         "id": "slf4j",
         "purl": "",
         "path": "https://github.com/qos-ch/slf4j.git",
-        "branch": "v_1.7.36",
+        "branch": "1.7-maintenance",
```

Comment:
How about leaving the branch empty? With the support added by this PR for checking out commits it should work, right?

Reply:
At the moment, providing a commit digest without providing a branch is not supported by Macaron, unfortunately. To be able to do that, we must also update the Analyzer class. 
I can certain update the logic to know allow providing a commit digest to checkout directly and remove the support for the case of `if both ``branch_name`` and a commit are provided, this function will checkout the commit directly only if that commit exists in the branch origin/<branch_name>. If not, this fucntion will return False.` However, I think it's out of scope for this PR. This PR was only intended to improve the git checkout operation, therefore I tried my best to keep the same behavior as before.
I suggest we can totally remove the branch name here in a separate PR. Please let me know what you think about it.

Reply:
No, we shouldn't remove support for the branches from the CLI. The purpose of this PR is a little confusing because on one hand a case is added in `check_out_repo_target` to allow checking out a commit without when a branch is not provided, but we can't actually test it because more changes are needed in the `Analyzer`. Can you please add that change in this PR? The goal is to allow checking out a commit without a branch being provided.

Reply:
Added in https://github.com/oracle/macaron/pull/548/commits/9db427c1a4889df7e18ff93f005288c20f73d5e3